### DISCUSSION TITLE
MediaRelay PiH264StreamTrack

### DISCUSSION THREAD

**Discussion by unitw** [aiortc/aiortc#655] [Q&A]
Over the past days i tried make my python picamera webrtc application run on a raspberry pi zero w 2 and hit a few roadblocks. 
The biggest one is the CPU usage besides the aviable hardware encoders.
i stumbled over a few attemps [ (Pull request 559)](https://github.com/aiortc/aiortc/pull/559) but i couldnt make it work.

I settled with the fork from [jpiat](https://github.com/jpiat) [aiortc](https://github.com/jpiat/aiortc) and the [PiH264StreamTrack](https://github.com/jpiat/aiortc/blob/main/examples/picamera/pitrack.py) implementation to send my video stream to the browser. 
The last hurdle now is relaying the picamera stream track when multiple peers trying to connect to the same picam.
```
    def create_pi_track(self):
        global relay, camera, webcam, videotrack
        if relay is None:
            videotrack = PiH264StreamTrack(30)
            camera = picamera.PiCamera()
            camera.resolution = (1280, 720)
            camera.framerate = 30
            target_bitrate = camera.resolution[0] * \
                             camera.resolution[1] * \
                             camera.framerate * 0.150
            camera.start_recording(
                videotrack,
                format="h264",
                profile="constrained",
                bitrate=int(target_bitrate),  # From wowza recommended settings
                inline_headers=True,
                sei=False,
            )
            relay = MediaRelay()
        return relay.subscribe(videotrack)

```
The shown approach doesnt seem to work. Any input on how to solve this?
Thanks for the continueing effort in improving this libary. 

PS
Maybe mention the raspberry pi camera topic  in the readme/docu/example, so that people dont struggle as much when they try to develop on raspian.

**Comment by hackernese** [comment]
Excuse me but the reference link to the [PiH264StreamTrack](https://github.com/jpiat/aiortc/blob/main/examples/picamera/pitrack.py) doesn't seem to work anymore, so do you happen to have its source code or how to import it ? i would like to try

**Reply by marcus2vinicius** [reply]
I would like to know too...
